examples/pebbleui/js/modules/PebbleUI.js

```diff
@@ -0,0 +1,561 @@
+    var keys = parent._keys;
+
+    if (code < 37 || code > 40) return;   // ignore everything except the arrow keys
+
+    delete keys[code];
+  });
+
+  // Everything is an Element!
+  this.Element = function(options) {
+    this.options = options || { };
+
+    this.bounds = this.options.bounds || [0, 0, 144, 168];
+    this._backgroundColor = options.backgroundColor || rocky.GColorClear;
+    this._color = options.color || rocky.GColorBlack;
+    this.z = options.z || 0;
+
+    // Intentionally empty
+    this._render = function(ctx, bounds) { };
+
+    return this;
+  };
+
+  // A Line is an Element
+  this.Line = function(options) {
+    options = options || {};
+
+    var lineElement = new parent.Element(options);
+    lineElement._p1 = [options.bounds[0], options.bounds[1]];
+    lineElement._p2 = [
+      options.bounds[0] + options.bounds[2],
+      options.bounds[1] + options.bounds[3]
+    ];
+    lineElement._strokeWidth = options.width || 2;
+
+    lineElement._render = function(ctx, bounds) {
+      rocky.graphics_context_set_stroke_color(ctx, this._color);
+      rocky.graphics_context_set_stroke_width(ctx, this._strokeWidth);
+      rocky.graphics_draw_line(ctx, this._p1, this._p2);
+    };
+
+    return lineElement;
+  };
+
+  // A Rect is an Element
+  this.Rect = function(options) {
+    options = options || {};
+
+    var rectElement = new parent.Element(options);
+
+    rectElement._render = function(ctx, bounds) {
+      rocky.graphics_context_set_fill_color(ctx, this._backgroundColor);
+      rocky.graphics_context_set_stroke_color(ctx, this._color);
+      rocky.graphics_fill_rect(ctx, this.bounds, 0, rocky.GCornersAll);
+    };
+
+    return rectElement;
+  };
+
+  // A Circle is an Element
+  this.Circle = function(options) {
+    options = options || {};
+
+    var circleElement = new parent.Element(options);
+    circleElement._render = function(ctx, bounds) {
+      rocky.graphics_context_set_fill_color(ctx, this._backgroundColor);
+      rocky.graphics_context_set_stroke_color(ctx, rocky.GColorClear);
+      rocky.graphics_fill_radial(ctx, this.bounds, rocky.GOvalScaleModeFitCircle,
+                                Math.min(this.bounds[2], this.bounds[3]) / 2,
+                                0, rocky.DEG_TO_TRIGANGLE(360));
+    };
+
+    return circleElement;
+  };
+
+  // A Text is an Element
+  this.Text = function(options) {
+    options = options || {};
+
+    var textElement = new parent.Element(options);
+
+    textElement._text = options.text || '';
+    textElement._font = rocky.fonts_get_system_font(options.font) ||
+                        rocky.fonts_get_system_font(parent._defaultFont);
+    textElement._color = options.color || rocky.GColorBlack;
+    textElement._alignment = options.alignment || rocky.GTextAlignmentLeft;
+
+    textElement.set = function(data) {
+      if (typeof data === 'string') {
+        this._text = data;
+      } else {
+        for (var key in data) {
```

Comment:
if you want to iterate over an array, you should consider using a different approach "for...in should not be used to iterate over an Array where the index order is important." https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in